src/polygon.h

```diff
@@ -180,14 +184,14 @@ class STriangle {
         Vector normals[3];
     };
 
-    static STriangle From(STriMeta meta, Vector a, Vector b, Vector c);
+    static STriangle From(const STriMeta& meta, const Vector& a, const Vector& b, const Vector& c);
```

Comment:
What's the cutoff? How big does a type need to be before it's worthwhile to pass by reference/pointer instead of by value? I can find a number of references online indicating that it's good practice in C++ to pass any non-primitive type by reference. Has this stuff been profiled?

Reply:
> Has this stuff been profiled?

Yes, and we've converted most functions on the hot paths to pass POD classes that are larger than a pointer to pass-by-const-reference.